Realm/Realm.SourceGenerator/Analytics.cs

```diff
@@ -0,0 +1,524 @@
+                    break;
+                case nameof(SdkFeature.WriteAsync):
+                    _realmFeaturesToAnalyse[SdkFeature.WriteAsync] = 1;
+                    break;
+                case nameof(SdkFeature.ThreadSafeReference):
+                    _realmFeaturesToAnalyse[SdkFeature.ThreadSafeReference] = 1;
+                    break;
+                case nameof(SdkFeature.FIXME_TWO):
+                    _realmFeaturesToAnalyse[SdkFeature.FIXME_TWO] = 1;
+                    break;
+                case nameof(SdkFeature.ShouldCompactOnLaunch):
+                    _realmFeaturesToAnalyse[SdkFeature.ShouldCompactOnLaunch] = 1;
+                    break;
+                case nameof(SdkFeature.MigrationCallback):
+                    _realmFeaturesToAnalyse[SdkFeature.MigrationCallback] = 1;
+                    break;
+                case nameof(SdkFeature.RealmChanged):
+                    _realmFeaturesToAnalyse[SdkFeature.RealmChanged] = 1;
+                    break;
+                case "SubscribeForNotifications":
+                    ITypeSymbol referenceType;
+                    try
+                    {
+                        // target.SubscribeForNotifications()
+                        // ^^^^^^--------------------------------------------(              )
+                        // of the SimpleMemberAccessExpression, get the first IdentifierName's type
+                        referenceType = context.SemanticModel.GetTypeInfo(identifierNameSyntax.Parent.ChildNodes().First()).Type;
+                    }
+                    catch (Exception ex)
+                    {
+                        Diagnostics.Add(SourceGenerator.Diagnostics.AnalyticsDebugInfo(
+                            $"{identifierNameSyntax} is likely some user defined syntax token.{Environment.NewLine}{ex.Message}"));
+                        break;
+                    }
+
+                    switch (referenceType.Name)
+                    {
+                        case "IQueryable":
+                        case "IOrderedQueryable":
+                            _realmFeaturesToAnalyse[SdkFeature.ResultSubscribeForNotifications] = 1;
+                            break;
+                        case "IList":
+                            _realmFeaturesToAnalyse[SdkFeature.ListSubscribeForNotifications] = 1;
+                            break;
+                        case "ISet":
+                            _realmFeaturesToAnalyse[SdkFeature.SetSubscribeForNotifications] = 1;
+                            break;
+                        case "IDictionary":
+                            _realmFeaturesToAnalyse[SdkFeature.DictionarySubscribeForNotifications] = 1;
+                            break;
+                        default:
+                            Diagnostics.Add(
+                                SourceGenerator.Diagnostics.AnalyticsDebugInfo($"{referenceType.Name} is not a collection type that is supported for notifications"));
```

Comment:
I suppose that this is only for debug, right?

Reply:
Correct. I was struggling with Diagnostics not showing so I filled the code with it. In the next iteration, that's only for `Debug.WriteLine`.